exporter/loggingexporter/logging_exporter.go

```diff
@@ -29,39 +29,53 @@ import (
 	"go.opentelemetry.io/collector/internal/otlptext"
 )
 
+var (
+	logsMarshaler    = otlptext.NewTextLogsMarshaler()
+	metricsMarshaler = otlptext.NewTextMetricsMarshaler()
+	tracesMarshaler  = otlptext.NewTextTracesMarshaler()
+)
+
 type loggingExporter struct {
 	logger *zap.Logger
 	debug  bool
 }
 
-func (s *loggingExporter) pushTraces(
-	_ context.Context,
-	td pdata.Traces,
-) error {
-
+func (s *loggingExporter) pushTraces(_ context.Context, td pdata.Traces) error {
 	s.logger.Info("TracesExporter", zap.Int("#spans", td.SpanCount()))
 
 	if !s.debug {
 		return nil
 	}
 
-	s.logger.Debug(otlptext.Traces(td))
-
+	// Ignore error since text Marshaler never returns error.
+	buf, _ := tracesMarshaler.Marshal(td)
```

Comment:
Doe we need the interface to have `error` in its return type if it never returns an error?

Reply:
The interface is generic "Marshaler" and will return error, for example for binary protobuf the proto.Marshaler returns error. In case of the text version there is no error.

Reply:
@bogdandrutu should it make assumptions about the underlying implementation? What if one day it does return error for whatever reason?

Reply:
And if it does what would we do with it? I don't have a strong opinion.

Reply:
> And if it does what would we do with it?

Just return the error right? This is in pushTraces that can returns `error` right? ie just the usual:

```go
buf, err := tracesMarshaler.Marshal(td)
if err != nil {
  return err
}
```

Reply:
Will do that then, unfortunately cannot test but will do a trick.